packages/explorer/src/app/(explorer)/hooks/useIndexerApiUrl.ts

```diff
@@ -0,0 +1,19 @@
+import { anvil, garnet, redstone } from "viem/chains";
+import { useChain } from "./useChain";
+
+const dozerUrl = {
+  [garnet.id]: "https://indexer.mud.garnetchain.com/q",
+  [redstone.id]: "https://indexer.mud.redstonechain.com/q",
+} as const;
+
+export function useIndexerApiUrl() {
+  const { id: chainId } = useChain();
+  if (chainId === anvil.id) {
+    return "/api/sqlite-indexer";
+  }
+
+  if (!(chainId in dozerUrl)) {
+    throw new Error(`Dozer URL not found for chain id ${chainId}`);
+  }
+  return dozerUrl[chainId as keyof typeof dozerUrl];
+}
```

Comment:
can we move this to a regular function rather than hook and have it return an object like
```
{ type: "sqlite", url: "/api/sqlite-indexer" }
```

```
{ type: "hosted", url: `${chain.indexerUrl}/q` }
```

then in other areas where we query each indexer, rather than checking the chain ID, we check the `type` to determine the format of the sql?

Reply:
good point, will adjust